ambry-messageformat/src/main/java/com.github.ambry.messageformat/MessageFormatSend.java

```diff
@@ -317,12 +320,13 @@ public int read(byte b[], int off, int len) throws IOException {
     if (off < 0 || len < 0 || len > b.length - off) {
       throw new IndexOutOfBoundsException();
     }
```

Comment:
Contract also says that 0 should be returned if `len == 0`. Should just be another `if` condition (and a check in the test).